pkg/api/types.go

```diff
@@ -551,6 +553,20 @@ type RBDVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// MetadataVolumeSource represents a volume containing metadata about a pod.
+type MetadataVolumeSource struct {
```

Comment:
Will this always be for metadata only, or do we plan to generalize it as we generalize the downward API? For instance, we're discussing downward API for resources #9473.

cc @pmorie